apps/iotune/iotune.cc

```diff
@@ -736,6 +743,8 @@ int main(int ac, char** av) {
             auto accuracy = configuration["accuracy"].as<unsigned>();
             auto saturation = configuration["saturation"].as<sstring>();
             auto random_io_buffer_size = configuration["random-io-buffer-size"].as<unsigned>();
+            auto& io_depth_conf = configuration["force-io-depth"];
+            std::optional<unsigned> force_io_depth = io_depth_conf.empty() ? std::nullopt : std::make_optional(io_depth_conf.as<unsigned>());
```

Comment:
Probably plain `unsigned` value will work if dedicating the value of `0` to "auto-detect" meaning. With that the declaration of "force-io-depth" parameter may have `->default_value(0)` and the whole thing gets much simpler and automatically handles the invalid `--force-io-depth 0` case

Reply:
Sure will change.

Reply:
Have pushed a change with the suggested change. Let me know what you think.